rclcpp/minimal_composition/compose_at_run_time.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdlib.h>
+#include <memory>
+#include <string>
+#include <vector>
+#include "class_loader/class_loader.h"
+#include "rclcpp/rclcpp.hpp"
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::executors::SingleThreadedExecutor exec;
+  std::vector<std::shared_ptr<rclcpp::Node>> nodes;  // save shared pointers !
+  std::vector<std::string> lib_names;
+  char * ament_prefix_path = getenv("AMENT_PREFIX_PATH");
```

Comment:
@codebot Btw. you should really compile the code with `--isolated` locally which you seem to be not doing.